dependency/vault_common.go

```diff
@@ -333,14 +333,15 @@ func isKVv2(client *api.Client, path string) (string, bool, error) {
 	return mountPath, false, nil
 }
 
-func addPrefixToVKVPath(p, mountPath, apiPrefix string) string {
+func addPrefixToVKVPath(p, mountPath, apiPrefixRaw string) string {
+	apiPrefix := path.Clean(apiPrefixRaw)
```

Comment:
Why Clean() here? The KV prefixes have a path like structure, they don't follow all the rules. Seems like it might have some unintended side effect.

Reply:
Thanks for the context on the prefix structure and good point, updated changes to make it more explicit on the trailing `/` check